### ISSUE TITLE
uts46_remap() fails in idna 3.9

### ISSUE THREAD

**Issue by lovetox** [kjd/idna#193]
```python
idna.uts46_remap("microsoft.com")
```

yields

`idna.core.InvalidCodepoint: Codepoint U+002E at position 10 does not follow STD3 rules`

so in essence i cannot put a domain name anymore as input to the `uts46_remap()` function.

is this expected with the change to Unicode 16?

I found no clear documentation what the input should be for this method, so i can only go after the argument naming, which states `domain` as input.

**Comment by lovetox** [comment]
I read a bit the standard, and as i understand it, STD3 validation must be done on labels and not on the full domain string.

**Comment by kjd** [comment]
The only publically document methods are `encode()` and `decode()`, with the remainder as helper functions which are underspecified. There were some changes in Unicode 16 to UTS46 that I am investigating and will likely release a newer version yet to fully implement, I am not sure if it is related yet.

**Comment by lovetox** [comment]
There should be no difference, `encode` simply calls `uts46_remap` in the first few lines with whatever string you pass to `encode`.

This is only related to unicode 16 in the sense that this was the point you choose to implement STD3 validation.
But validation needs to be done on labels not on the domain.

Just to make the impact of this change clear, every use of 

```python
idna.encode(domain, std3_rules=True)
```

is broken.

Thanks

**Comment by kjd** [comment]
I just released v3.10 which reverts back to Unicode 15.1 which fixes this issue. The UTS46 implementation needs some significant work to address the changes in Unicode 16, will re-release later once it fixed. This surfaced due to an unrelated patch that meant the entire test suite wasn't being properly executed, as described in issue #192; so running the unit tests didn't evidence this prior to release.
